packages/tracing/src/integrations/node/ioredis.ts

```diff
@@ -0,0 +1,101 @@
+import { Hub } from '@sentry/hub';
+import { EventProcessor, Integration, SpanContext } from '@sentry/types';
+import { fill, loadModule, logger } from '@sentry/utils';
+
+type CommandArgs = string | Buffer | number | unknown[];
+
+interface Command {
+  name: string;
+  args: CommandArgs[];
+}
+
+interface IORedisInstance {
+  prototype: {
+    sendCommand: (command: Command, ...args: unknown[]) => unknown;
+  };
+}
+
+interface IORedisOptions {
+  // location of the ioredis package
+  // allows users to specify the direct in case they are using
+  // libraries that are wrappers around ioredis
+  moduleLocation?: string;
+}
+
+/** Tracing integration for the IORedis library */
+export class IORedis implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'IORedis';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = IORedis.id;
+
+  /**
+   * Location of the IORedis package
+   */
+  private _moduleLocation: string;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options: IORedisOptions = {}) {
+    this._moduleLocation = options.moduleLocation ?? 'ioredis';
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    const pkg = loadModule<IORedisInstance>(this._moduleLocation);
+
+    if (!pkg) {
+      logger.error(`IORedis integration was unable to require \`${this._moduleLocation}\` package.`);
+      return;
+    }
+
+    this._patchOperation(pkg, getCurrentHub);
+  }
+
+  /**
+   *  Patches the sendCommand function to utilize tracing
+   */
+  private _patchOperation(ioredis: IORedisInstance, getCurrentHub: () => Hub): void {
+    const getSpanContext = this._getSpanContextFromCommandArguments.bind(this);
+
+    fill(ioredis.prototype, 'sendCommand', function(orig: () => Promise<unknown>) {
+      return function(this: unknown, command: Command, ...args: unknown[]) {
+        const scope = getCurrentHub().getScope();
+        const parentSpan = scope?.getSpan();
+
+        const span = parentSpan?.startChild(getSpanContext(command));
```

Comment:
Can we just do

```suggestion
        const span = parentSpan?.startChild({
          op: 'redis',
          description: command.name,
          data: {
            arguments: command.args.toString(),
          },
        });
```

and remove the `_getSpanContextFromCommandArguments()` function? We don't have the special cases that `mongo` does here, no need to add the complexity of another function call then.